### PR TITLE
integration-tests/smoke/ccip: increase batch size

### PR DESCRIPTION
**Pull request by makramkd**
Ticket: https://smartcontract-it.atlassian.net/browse/CCIP-4398

* Switch to in-memory test instead of docker to reduce test duration
* Split tests into separate `Test_` tests instead of `t.Run` for more robustness and isolation
* Axe the sendEth logic and just use the `Users` set in the `Environment` which should already be funded

Requires:
- https://github.com/smartcontractkit/chainlink-ccip/pull/327

### CODE REVIEW

**Review by jmank88 — DISMISSED** [review]

File: integration-tests/smoke/ccip/ccip_batching_test.go [hunk 1]
```diff
@@ -373,18 +394,27 @@ func sendMessagesAsync(
...
-		e.Env.Chains[sourceChainSelector].DeployerKey,
-		state.Chains[sourceChainSelector].OnRamp,
-		state.Chains[sourceChainSelector].Router,
-		state.Chains[sourceChainSelector].Multicall3,
-		destChainSelector,
-		numMessages,
-		common.LeftPadBytes(state.Chains[destChainSelector].Receiver.Address().Bytes(), 32),
-	)
+	var err error
+	for i := 0; i < 3; i++ {
```
Comment (dimkouv) [inline integration-tests/smoke/ccip/ccip_batching_test.go]: better define `3` in a const.

Comment (makramkd) [reply integration-tests/smoke/ccip/ccip_batching_test.go]: Thx, fixed

**Review by dimkouv — DISMISSED** [review]

File: deployment/environment/memory/chain.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ func evmChain(t *testing.T, numUsers int) EVMChain {
 	owner, err := bind.NewKeyedTransactorWithChainID(key, big.NewInt(1337))
 	require.NoError(t, err)
 	genesis := types.GenesisAlloc{
-		owner.From: {Balance: big.NewInt(0).Mul(big.NewInt(700000), big.NewInt(params.Ether))}}
+		owner.From: {Balance: assets.Ether(100_000).ToInt()}}
```
Comment (asoliman92) [inline deployment/environment/memory/chain.go]: Why lowering the amount?

**Review by asoliman92 — DISMISSED** [review]

File: deployment/environment/memory/chain.go (hunk 2 above)
Comment (makramkd) [reply deployment/environment/memory/chain.go]: Number dyslexic!

Comment (makramkd) [reply deployment/environment/memory/chain.go]: Thats also why I added the underscores

**Review by dimkouv — APPROVED** [review]

**Review by asoliman92 — APPROVED** [review]

**Review by mateusz-sekara — APPROVED** [review]

**Review by b-gopalswami — APPROVED** [review]

**Review by AnieeG — APPROVED** [review]

**Review by kalverra — APPROVED** [review]

**Pull request merged**